commit 3459b5b0f98ab7e353f359942bb77af1319c98ce
Author: Trent Mick <trentm@gmail.com>
Date:   2019-04-12T13:07:02-07:00 (6 months ago)
    
    TRITON-1403 drop cloudapi "test mode" (aka config.test, CLOUDAPI_TEST_MODE)

diff --git a/lib/app.js b/lib/app.js
index 5072b1be9..fd4bd981d 100644
--- a/lib/app.js
+++ b/lib/app.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -55,7 +55,6 @@ var networkMiddleware = require('./middleware/networks');
 var audit = require('./audit');
 var auditLogger = require('./audit_logger');
 var rules = require('./rules');
-var tests = require('./tests');
 var volumeEndpoints = require('./endpoints/volumes');
 
 // Account users, roles and policies:
@@ -559,10 +558,6 @@ module.exports = {
                 volumeEndpoints.mount(server, userThrottle(config, 'volumes'));
             }
 
-            if (config.test) {
-                tests.mount(server, emptyThrottle(), config);
-            }
-
             server.on('after', auditLogger({
                 log: log.child({component: 'audit'})
             }));
diff --git a/lib/machines.js b/lib/machines.js
index 24fa1e86e..eda4ffa78 100644
--- a/lib/machines.js
+++ b/lib/machines.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2019, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -843,20 +843,6 @@ function getCreateOptions(req) {
         opts.nic_driver = img.nic_driver;
     }
 
-    // Intentionally not documented, at least until we are checking on
-    // vmapi that owner is allowed to specify a given server:
-    // PUBAPI-724: Only allow in test mode
-    if (params.server_uuid && req.config.test) {
-        opts.server_uuid = params.server_uuid;
-    }
-
-    // Another alternative to provide server_uuid, (which I mostly use
-    // locally to run node-smartdc tests w/o having to hardcode server_uuid
-    // into sdc-createmachine):
-    if (process.env.SERVER_UUID) {
-        opts.server_uuid = process.env.SERVER_UUID;
-    }
-
     if (params.locality) {
         opts.locality = params.locality;
     }
diff --git a/lib/tests.js b/lib/tests.js
deleted file mode 100644
index 4bd177e21..000000000
--- a/lib/tests.js
+++ /dev/null
@@ -1,46 +0,0 @@
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2016, Joyent, Inc.
- */
-
-/*
- * This file contains functions useful during the testing of Cloudapi. They
- * are only active when the configuration has Cloudapi in testing mode.
- */
-
-var assert = require('assert-plus');
-
-
-function throwException(req, res, next) {
-    // non-existent function invocation should throw exception
-    req.asdasdadasdasd();
-    next();
-}
-
-
-function mount(server, before, config) {
-    assert.object(server);
-    assert.ok(before);
-    assert.ok(config);
-
-    if (!config.test) {
-        return server;
-    }
-
-    server.get({
-        path: '/:account/tests/throw_exception',
-        name: 'ThrowException'
-    }, before, throwException);
-
-    return server;
-}
-
-
-module.exports = {
-    mount: mount
-};
diff --git a/sapi_manifests/cloudapi/template b/sapi_manifests/cloudapi/template
index 703da3f1a..71794e444 100644
--- a/sapi_manifests/cloudapi/template
+++ b/sapi_manifests/cloudapi/template
@@ -158,19 +158,6 @@
     {{/account_allowed_dcs}}
     "account_allowed_dcs_msg": "{{{account_allowed_dcs_msg}}}",
 
-    "bleeding_edge_features": {
-        {{#CLOUDAPI_BLEEDING_EDGE_FEATURES}}
-        "{{{.}}}": true,
-        {{/CLOUDAPI_BLEEDING_EDGE_FEATURES}}
-        "": false
-    },
-    "bleeding_edge_login_whitelist": {
-        {{#CLOUDAPI_BLEEDING_EDGE_LOGIN_WHITELIST}}
-        "{{{.}}}": true,
-        {{/CLOUDAPI_BLEEDING_EDGE_LOGIN_WHITELIST}}
-        "": false
-    },
-
     {{^fabric_cfg}}
     "fabrics_enabled": false,
     {{/fabric_cfg}}
@@ -198,10 +185,16 @@
     "experimental_cloudapi_automount_nfs_shared_volumes": {{{experimental_cloudapi_automount_nfs_shared_volumes}}},
     {{/experimental_cloudapi_automount_nfs_shared_volumes}}
 
-    {{^CLOUDAPI_TEST_MODE}}
-    "test": false
-    {{/CLOUDAPI_TEST_MODE}}
-    {{#CLOUDAPI_TEST_MODE}}
-    "test": {{{CLOUDAPI_TEST_MODE}}}
-    {{/CLOUDAPI_TEST_MODE}}
+    "bleeding_edge_features": {
+        {{#CLOUDAPI_BLEEDING_EDGE_FEATURES}}
+        "{{{.}}}": true,
+        {{/CLOUDAPI_BLEEDING_EDGE_FEATURES}}
+        "": false
+    },
+    "bleeding_edge_login_whitelist": {
+        {{#CLOUDAPI_BLEEDING_EDGE_LOGIN_WHITELIST}}
+        "{{{.}}}": true,
+        {{/CLOUDAPI_BLEEDING_EDGE_LOGIN_WHITELIST}}
+        "": false
+    }
 }
diff --git a/test/common.js b/test/common.js
index 1617ab030..c3b46dd41 100644
--- a/test/common.js
+++ b/test/common.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 /*
@@ -511,8 +511,6 @@ function loadServer(cb) {
         return cb(null, serverObj);
     }
 
-    CONFIG.test = true;
-
     return app.createServer(CONFIG, function (err, server) {
         if (err) {
             return cb(err);
diff --git a/test/runtests b/test/runtests
index 650725b5c..70f852944 100755
--- a/test/runtests
+++ b/test/runtests
@@ -6,7 +6,7 @@
 #
 
 #
-# Copyright (c) 2015, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -99,10 +99,6 @@ echo "# Setup a clean output dir ($OUTPUT_DIR)."
 rm -rf /var/tmp/cloudapitest
 mkdir -p /var/tmp/cloudapitest
 
-# Gather DC setup info for the test files.
-echo "# Datacenter config:"
-echo "#     SDC_SSO_ADMIN_IP is $SDC_SSO_ADMIN_IP"
-
 cd $TOP
 
 # Generate list of tests
diff --git a/test/tests.test.js b/test/tests.test.js
deleted file mode 100644
index 059512b70..000000000
--- a/test/tests.test.js
+++ /dev/null
@@ -1,65 +0,0 @@
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2018, Joyent, Inc.
- */
-
-var test = require('@smaller/tap').test;
-var common = require('./common');
-
-
-// --- Globals
-
-
-var CLIENTS;
-var CLIENT;
-var SERVER;
-var TEST;
-
-
-// --- Tests
-
-
-test('setup', function (t) {
-    common.setup(function (_, clients, server) {
-        CLIENTS = clients;
-        CLIENT  = clients.user;
-        SERVER  = server;
-
-        TEST = common.getCfg().test;
-
-        t.end();
-    });
-});
-
-
-test('uncaughtException handler OK', function (t) {
-    if (!TEST) {
-        return t.end();
-    }
-
-    return CLIENT.get('/my/tests/throw_exception',
-            function (err, req, res, body) {
-        t.ok(err);
-        t.equal(res.statusCode, 500);
-
-        t.deepEqual(body, {
-            code: 'InternalError',
-            message: 'Internal Error'
-        });
-
-        t.end();
-    });
-});
-
-
-test('teardown', function (t) {
-    common.teardown(CLIENTS, SERVER, function (err) {
-        t.ifError(err, 'teardown success');
-        t.end();
-    });
-});
